### PR TITLE
T591397: NumberBox does not work if format option is defined, current locale is "ar" and Intl is used

### PR DESCRIPTION
**Pull request by mrjohn0011**

### CODE REVIEW

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js [hunk 1]
```diff
@@ -374,6 +375,22 @@ QUnit.test("boundary value should correctly apply after second try to set overfl
 
 QUnit.module("format: text input", moduleConfig);
 
+QUnit.test("input should not be prevented in arabic locale", function(assert) {
+    var originalCulture = localization.locale();
+
+    try {
+        localization.locale("ar");
```
Comment (nightskylark) [inline testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js]: What is the point to set Arabic locale if none localization engine is connected? Maybe it's better to mock `numberLocalization.format` method for this test?

**Review by nightskylark — CHANGES_REQUESTED** [review]
